contrib/intel/jenkins/summary.py

```diff
@@ -7,389 +8,493 @@
+            self.fails += 1
 
-    print_results(f"{prov} fabtests {build_mode}", passes, fails, failed_tests)
-
-    log.close()
-    return int(fails)
-
-def summarize_fabtests(log_dir, prov, build_mode=None):
-    file_name = f'{prov}_fabtests_{build_mode}'
-    if not os.path.exists(f'{log_dir}/{file_name}'):
+    def check_exclude(self):
         return
 
-    log = open(f'{log_dir}/{file_name}', 'r')
-    line = log.readline()
-    passes = 0
-    fails = 0
-    excludes = 0
-    failed_tests = []
-    excluded_tests = []
-    test_name_string='no_test'
-    while line:
+    def check_line(self):
+        self.check_name()
+        self.check_pass()
+        self.check_fail()
+        self.check_exclude()
+
+    def read_file(self):
+        self.log = open(self.file_path, 'r')
+        self.line = self.log.readline().lower()
+        while self.line:
+            self.check_line()
+            self.line = self.log.readline().lower()
+
+        self.log.close()
+
+    def summarize(self):
+        if not self.exists:
+            return 0
+
+        self.read_file()
+        self.print_results()
+        return int(self.fails)
+
+class FiInfoSummarizer(Summarizer):
+    def __init__(self, log_dir, prov, build_mode):
+        super().__init__(log_dir, prov, build_mode,
+                         f'{prov}_fi_info_{build_mode}',
+                         f"{prov} fi_info {build_mode}")
+
+    def check_fail(self):
+        if "exiting with" in self.line:
+            self.fails += 1
+            self.failed_tests.append(f"fi_info {self.prov}")
+
+    def read_file(self):
+        self.log = open(self.file_name, 'r')
+        line = self.log.readline()
+        while line:
+            self.check_line()
+            line = self.log.readline()
+
+        if not self.fails:
+            self.passes += 1
+
+        self.log.close()
+
+class FabtestsSummarizer(Summarizer):
+    def __init__(self, log_dir, prov, build_mode):
+        super().__init__(log_dir, prov, build_mode,
+                         f'{prov}_fabtests_{mode}',
+                         f"{prov} fabtests {mode}")
+        self.result = ''
+        self.result_line = ''
+
+    def check_name(self):
         # don't double count ubertest output
-        if 'ubertest' in line and 'client_cmd:' in line:
-            while 'name:' not in line: # skip past client output in ubertest
-                line = log.readline()
+        if 'ubertest' in self.line and 'client_cmd:' in self.line:
+            while 'name:' not in self.line: # skip past client output in ubertest
+                self.line = self.log.readline().lower()
 
-        if 'name:' in line:
-            test_name = line.split()[2:]
-            test_name_string = ' '.join(test_name)
+        if 'name:' in self.line:
+            test_name = self.line.split()[2:]
+            self.test_name_string = ' '.join(test_name)
 
-        if 'result:' in line:
-            result_line = line.split()
+    def get_result_line(self):
```

Comment:
get_result_line seems to be used to parse the line for the result, and then the caller of get_result_line is also resetting self.result.

instead, you could just return the result and not store it, then you don't have to remember to reset it.  additionally, it seems some places want to parse the full matching line, in which case you could change this function to return a tuple

```
def get_result_line(self, line) -> Tuple[str,str]:
    result = line.split("result:")
    if len(result) > 1:
        return (result[-1].lower().strip(), line)
    return (None, None)
```

then the callers would do something like
```
result, result_line = self.get_result_line(line)
```
or `result, _ = self.get_result_line(line)` if the result_line is unused